### ISSUE TITLE
Does opacus work on masked language models task?

### ISSUE THREAD

**Issue by Jasper-Dong** [pytorch/opacus#333]
## 🚀 Feature
Thanks for the tutorial of **building_text_classifier.ipynb**. In this tutorial, a BERT model used for Sequence Classification is adopted. I tried to build a Masked Language Model with opacus, but there are a few issues. So is that possible to adapt opacus to Masked Language Model?

## Motivation

Because I am trying to explore language models on multiple tasks including, Text Classification, Text Classification, and Fill Mask. Meanwhile, I also want to adapt opacus compared with normal models. So it would be great if you can support my work. Thank you.

**Comment by pierrestock** [comment]
Hey Jasper-Dong,

Thanks for your interst in Opacus. Could you elaborate on the type of issues you encounter? For instance, BERT is usually pre-trained with a MLM objective. 

If you wish to change the downstream task, you can rely on the tutorial and only change (1) the loss function and (2) the last(s) layers to match your downstream task's objective.

**Comment by Jasper-Dong** [comment]
Hi Pierre,

Thanks for your reply. Yes, the Masked Language Modeling that I meant is a downstream task, which is to use BERT for language modeling tasks. Specifically, I loaded the model from _model = AutoModelForMaskedLM.from_pretrained(model_checkpoint)_, and the last layer is _cls_:

_BertOnlyMLMHead(
  (predictions): BertLMPredictionHead(
    (transform): BertPredictionHeadTransform(
      (dense): Linear(in_features=768, out_features=768, bias=True)
      (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)
    )
    (decoder): Linear(in_features=768, out_features=105879, bias=True)
  )
)_

According to the tutorial, the layers before the last BERT layer are frozen. I did that too for this model, but the model performance is quite bad compared to normal optimizer. So have you tried this kind of task before or do you have any reasons for this?

Thanks for your help.

**Comment by ffuuugor** [comment]
Closing the task -  seemingly there's nothing wrong with opacus behaviour and no change is required. But adding DP is hard and more research is needed indeed.

If you haven't seen this paper, it discusses MLM task with DP-SGD and offers good baselines and techniques to achieve them: https://arxiv.org/pdf/2108.01624.pdf
